### PR TITLE
feat(graphiql): Prettify also formats query variables

### LINKED ISSUE #990: Unable to prettify query variables input
As a user it is really useful to prettify the input query/mutation/suscription. But AFAIK there's currently no way to prettify the query variables section.
If one follows the convention for mutations to receive a unique input field, it becomes cumbersome when a mutation has many arguments as one has to indent them manually.

image

### PR DESCRIPTION
**Pull request by zouxuoz**
Closes #990 

and rid of unnecessary editors updates when content not changed during prettification

### CODE REVIEW

**Comment by zouxuoz** [comment]
@benjie should I add some tests to increase coverage?)

File: packages/graphiql/src/components/GraphiQL.js [hunk 1]
```diff
@@ -758,7 +758,29 @@ export class GraphiQL extends React.Component {
...
+      prettifiedVariableEditorContent = JSON.stringify(
+        JSON.parse(variableEditorContent),
+        null,
+        2
+      );
+    } catch {}
+
+    if (prettifiedVariableEditorContent !== variableEditorContent) {
+      variableEditor.setValue(prettifiedVariableEditorContent);
+    }
```
Comment (benjie) [inline packages/graphiql/src/components/GraphiQL.js]: This'll clear variables if they weren't valid; let's not do that. Instead move it all into the `try` block:

```js
try {
  const prettifiedVariableEditorContent = JSON.stringify(JSON.parse(variableEditorContent), null, 2);
  if (prettifiedVariableEditorContent !== variableEditorContent) {
    variableEditor.setValue(prettifiedVariableEditorContent);
  }
} catch {
  /* Parsing JSON failed, skip prettification */
}
```

File: packages/graphiql/cypress/integration/prettify.spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const uglyVaribles = `{"a": 1}`;
+
+const brokenQuery = `longDescriptionType {id}}`;
+
+const brokenVaribles = `"a": 1}`;
+
+describe('GraphiQL Prettify', function() {
+  it('Regular prettification', function() {
+    cy.visit(`/?query=${uglyQuery}&variables=${uglyVaribles}`);
```
Comment (benjie) [inline packages/graphiql/cypress/integration/prettify.spec.js]: Please escape the URLs; it doesn't seem to matter for these tests but it'll help future tests get it right :+1: 

```suggestion
    cy.visit(`/?query=${encodeURIComponent(uglyQuery)}&variables=${encodeURIComponent(uglyVaribles)}`);
```

File: packages/graphiql/cypress/integration/prettify.spec.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    cy.visit(`/?query=${uglyQuery}&variables=${brokenVaribles}`);
+
+    cy.get('[title="Prettify Query (Shift-Ctrl-P)"]').click();
+
+    cy.window().then((w) => {
+      cy.expect(w.g.getQueryEditor().getValue()).to.equal(prettifiedQuery);
+      cy.expect(w.g.getVariableEditor().getValue()).to.equal(brokenVaribles);
+    })
+  }) 
+})
```
Comment (benjie) [inline packages/graphiql/cypress/integration/prettify.spec.js]: Excellent job of translating my list into tests! :raised_hands:

**Comment by zouxuoz** [comment]
@benjie Updated 🥳

**Comment by benjie** [comment]
Merged! Thanks :tada:

**Pull request merged**